include/openssl/crypto.h.in

```diff
@@ -85,10 +85,17 @@ __owur int CRYPTO_THREAD_write_lock(CRYPTO_RWLOCK *lock);
 int CRYPTO_THREAD_unlock(CRYPTO_RWLOCK *lock);
 void CRYPTO_THREAD_lock_free(CRYPTO_RWLOCK *lock);
 
+# ifdef __GNUC__
+typedef __attribute__((__aligned__(8))) uint64_t CRYPTO_ATOMIC_U64;
+# else
+typedef uint64_t CRYPTO_ATOMIC_U64;
+# endif
+
 int CRYPTO_atomic_add(int *val, int amount, int *ret, CRYPTO_RWLOCK *lock);
-int CRYPTO_atomic_or(uint64_t *val, uint64_t op, uint64_t *ret,
+int CRYPTO_atomic_or(CRYPTO_ATOMIC_U64 *val, uint64_t op, CRYPTO_ATOMIC_U64 *ret,
```

Comment:
This changes the signature of a public API function which we cannot do.

Reply:
ACK